pysatCDAAC/instruments/cosmic_gps.py

```diff
@@ -314,54 +334,212 @@ def load(fnames, tag=None, inst_id=None, altitude_bin=None):
         Object containing metadata such as column names and units
 
     """
-    global lower_l1_tags
 
-    # input check
+    # Input check.
     if altitude_bin is not None:
-        if tag != 'ionprf':
-            estr = 'altitude_bin keyword only supported for "tag=ionprf"'
+        if tag in ['podtec', 'scnlv1', 'ionphs']:
+            estr = ' '.join(['altitude_bin keyword only supported if `MSL_alt`',
+                             'present in the file.'])
             raise ValueError(estr)
 
     num = len(fnames)
-    # make sure there are files to read
+    # Make sure there are files to read.
     if num != 0:
-        # call separate load_files routine, segmented for possible
-        # multiprocessor load, not included and only benefits about 20%
-        output = pds.DataFrame(load_files(fnames, tag=tag, inst_id=inst_id,
-                                          altitude_bin=altitude_bin))
+
+        # Set up loading files with a mixture of data lengths.
+        if tag == 'atmprf':
+            coords = {}
+            temp_keys = ['OL_vec2', 'OL_vec1', 'OL_vec3', 'OL_vec4']
+            dim_label = 'dim1'
+            for key in temp_keys:
+                coords[key] = dim_label
+
+            temp_keys = ['OL_ipar', 'OL_par', 'ies', 'hes', 'wes']
+            dim_label = 'dim2'
+            for key in temp_keys:
+                coords[key] = dim_label
+        else:
+            # All other files have a single 2D dimension
```

Comment:
```suggestion
            # All other files have a single 2D dimension
```

I am confused by this.  Do all the other files have only one variable that is 2D?  Because that's not what the comment is saying.

Reply:
Looking at the files, I think it refers to possible values for the 2nd dimension. For atmprf:
```
Dimensions:           (time: 1598, RO: 2901, dim2: 20, dim1: 10249)
```
with data values potentially being of the shape (time, RO), (time, dim1), or (time, dim2).

for ionprf:
```
Dimensions:           (time: 1199, RO: 669)
```
I'll clean up and clarify.

Reply:
Also restructured the code to flow a little better while reading. https://github.com/pysat/pysatCDAAC/pull/40/commits/f95bd8f42e5136542304de009729c2974a403381